### ISSUE TITLE
[🐛 Bug]: Failure in onPrepare does not stop execution

### ISSUE THREAD

**Issue by leonard-mustatea** [webdriverio/webdriverio#8013]
### Have you read the Contributing Guidelines on issues?

- [X] I have read the [Contributing Guidelines on issues](https://github.com/webdriverio/webdriverio/blob/main/CONTRIBUTING.md#reporting-new-issues).

### WebdriverIO Version

7.16.13

### Node.js Version

v16.13.1

### Mode

WDIO Testrunner

### Which capabilities are you using?

```typescript
{
   browserName: 'chrome'.
   browserVersion: 'latest'
   platformName: 'MacOS Monteray'
}
```

### What happened?

Reference: https://github.com/webdriverio/webdriverio/issues/5398

It was nice seeing that there's a way of stopping test execution if a precondition isn't met inside a hook.
Unfortunately, at least in this version that functionality isn't working as expected.

Throwing SevereServiceError inside onPrepare hook doesn't stop test runner execution.

### What is your expected behavior?

If I deliberately throw SevereServiceError inside a hook the test run and process should be immediately terminated.

### How to reproduce the bug.

Add 
`throw new SevereServiceError(
                    `Please stop my test execution!`,
                );`
inside `onPrepare` or any other hook and execute your test command.

NOTE: The onPrepare hook fails, but does not stop the test runner from executing against the tests.

### Relevant log output

```typescript
ERROR @wdio/utils:shim: SevereServiceError: Please stop my test execution!
[0-2]     at Object.beforeSession (/Users/user/path/runner/hooks/hooks.browser.conf.ts:97:19)
[0-2]     at /Users/user/path/node_modules/@wdio/utils/build/shim.js:82:27
[0-2]     at new Promise (<anonymous>)
[0-2]     at /Users/user/path/node_modules/@wdio/utils/build/shim.js:79:47
[0-2]     at Array.map (<anonymous>)
[0-2]     at executeHooksWithArgsShim (/Users/user/path/node_modules/@wdio/utils/build/shim.js:79:33)
[0-2]     at Runner.run (/Users/user/path/node_modules/@wdio/runner/build/index.js:77:48)
[0-2]     at processTicksAndRejections (node:internal/process/task_queues:96:5)
[0-2] RUNNING in chrome - /src/tests/features/smoke/smoke.login.feature
[0-2] setDefinitionFunctionWrapper is deprecated and will be removed in version 8.0.0 of cucumber-js. If this was used to wrap generator functions, please transition to using async / await. If this was used to wrap step definitions, please use BeforeStep / AfterStep hooks instead. If you had other use cases, please create an issue.
[0-2] 2022-02-14 16:27:46.707  INFO  @>>> FEATURE: [feature][START] 
[0-2] 2022-02-14 16:27:46.724  INFO  @>>> SCENARIO: [scenario][START] 
[0-2] 2022-02-14 16:27:46.727  INFO  @>>> STEP: [Given I throw SevereServiceError] 
[0-2] 2022-02-14 16:27:49.385  INFO  Waiting for page / element to load; awaiting for the following element ul[class*='class'] > li to be displayed...
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Is there an existing issue for this?

- [X] I have searched the existing issues

**Comment by christian-bromann** [comment]
Thanks for reporting!

Currently this is only supported for services and not for hooks defined in the wdio config. I think we should make the behavior the same for both situations (even though you could just do a `process.exit(1)` to kill the process. Please check the [`runLauncherHook`](https://github.com/webdriverio/webdriverio/blob/main/packages/wdio-cli/src/utils.ts#L61) function, a similar behavior as in `runServiceHook` should be implemented. Maybe there is a way to merge both functions into one.

Any contributions that resolves the bug are highly appreciated. Please take a look into our [contribution guidelines](https://webdriver.io/docs/contribute) and let us know if you have any questions. Cheers!

**Comment by christian-bromann** [comment]
Fixed by #8100
